From 9216caf1a7610d0f4c66a27e90f2db302b8529fd Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 2 Jan 2025 18:13:47 +0100
Subject: [PATCH 0136/1013] dovetail: fs: rename oob_data to f_oob_ctx in
 struct file

The notion of "oob data" is used across several kernel subsystems,
particularly in the networking area. Dovetail's notion of out-of-band
relates to the execution context of a CPU on the high priority stage
of the interrupt pipeline instead (i.e. the one which runs
asynchronously to the regular scheduling of operations).

Mitigate the confusion in struct file by moving from the way too
generic "data" to "context", abiding by the common naming scheme there
in the same move, i.e. s/oob_data/f_oob_ctx/.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/linux/fs.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/linux/fs.h b/include/linux/fs.h
index bb0b1e433b38..78108527b2c1 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -1067,7 +1067,9 @@ struct file {
 		freeptr_t		f_freeptr;
 	};
 	/* --- cacheline 3 boundary (192 bytes) --- */
-	void			       *oob_data;
+#ifdef CONFIG_DOVETAIL
+	void				*f_oob_ctx;
+#endif
 } __randomize_layout
   __attribute__((aligned(4)));	/* lest something weird decides that 2 is OK */
 
-- 
2.30.2

